types/transaction_helpers.go

```diff
@@ -0,0 +1,110 @@
+// +build testing
+
+package types
+
+import (
+	"errors"
+)
+
+// TransactionGraph will return a set of valid transactions that all spend
+// outputs according to the input graph. Each [source, dest] pair defines an
+// edge of the graph. The graph must be fully connected and the granparent of
+// the graph must be the sourceOutput. '0' refers to an edge from the source
+// output. Each edge also specifies a value for the output, and an amount of
+// fees. If the fees are zero, no fees will be added for that edge. 'sources'
+// must be sorted.
+//
+// Example of acceptable input:
+//
+// sourceOutput: // a valid siacoin output spending to UnlockConditions{}.UnlockHash()
+//
+// Sources: [0, 0, 1, 2, 3, 3, 3, 4]
+// Dests:   [1, 2, 3, 3, 4, 4, 5, 6]
+//
+// Resulting Graph:
+//
+//    o
+//   / \
+//  o   o
+//   \ /
+//    o
+//   /|\
+//   \| \
+//    o  x // 'x' transactions are symbolic, not actually created
+//    |
+//    x
+//
+func TransactionGraph(sourceOutput SiacoinOutputID, sources []int, dests []int, outputValues []Currency, fees []Currency) ([]Transaction, error) {
```

Comment:
thinking about it. Then we get another type in our godoc, which I'm not super happy with. But 4 slices that are all tuples is pretty gross too.

4 slices has been easy enough to work with, though it produces pretty ugly code.